File: app/src/main/java/com/cyl/musiclake/ui/music/local/contract/ArtistSongContract.java
package com.cyl.musiclake.ui.music.local.contract;

import android.graphics.Bitmap;

import com.cyl.musiclake.bean.Music;
import com.cyl.musiclake.ui.base.BaseContract;

import java.util.List;

public interface ArtistSongContract {

    interface View extends BaseContract.BaseView {

        void showEmptyView();

        void showSongs(List<Music> songList);

        void showAlbumArt(Bitmap bitmap);
    }

    interface Presenter extends BaseContract.BasePresenter<View> {

        void loadSongs(String artistName);
    }

}

File: app/src/main/java/com/cyl/musiclake/ui/music/artist/contract/ArtistSongContract.kt
package com.cyl.musiclake.ui.music.artist.contract

import com.music.lake.musiclib.bean.BaseMusicInfo
import com.cyl.musiclake.ui.base.BaseContract

interface ArtistSongContract {

    interface View : BaseContract.BaseView {

        fun showEmptyView()

        fun showSongs(songList: MutableList<BaseMusicInfo>)
    }

    interface Presenter : BaseContract.BasePresenter<View> {

        fun loadSongs(artistName: String)
    }

}
